File: FoodOrderingApp/src/models/order.with.meal.data.response.ts
import { OrderItemWithMealResponse } from "./order.item.with.meal.response";

export interface OrderWithMealDataResponse {
     orderNumber: string;
      totalPrice: number;
      status: string;
      customerEmail: string;
      orderDate: string;
      orderItems: OrderItemWithMealResponse[];
      isCancellable: boolean;
}

File: services/order-service/src/main/java/com/example/orderservice/api/dto/OrderWithMealDataResponse.java
package com.example.orderservice.api.dto;

import java.math.BigDecimal;
import java.util.List;

public record OrderWithMealDataResponse(
        String orderNumber,
        BigDecimal totalPrice,
        String status,
        String customerEmail,
        String orderDate,
        List<OrderItemWithMealResponse> orderItems,
        boolean isCancellable
) {
}
